atomate/vasp/firetasks/exchange.py

```diff
@@ -0,0 +1,336 @@
+        else:
+            # Check for J_ij convergence
+            converged_list = []
+            for cutoff, hmodel in zip(cutoffs, hmodels):
+                ex_params = hmodel.ex_params
+                converged = True
+
+                # J_ij exists
+                if len(ex_params) > 1:
+                    E0 = abs(ex_params["E0"])
+                    for k, v in ex_params.items():
+                        if k != "E0":
+                            if abs(v) > E0:  # |J_ij| > |E0| unphysical!
+                                converged = False
+                else:  # Only <J> was computed
+                    converged = False
+
+                if converged:
+                    converged_list.append(hmodel.as_dict())
+
+            # If multiple Heisenberg Models converged, take the maximal cutoff
+            if len(converged_list) > 0:
+                hmodel = converged_list[-1]  # Largest cutoff
+            else:  # No converged models
+                hmodel = None
+
+        # Update FW spec with converged hmodel or None
+        update_spec = {"converged_heisenberg_model": hmodel}
+
+        return FWAction(update_spec=update_spec)
+
+
+@explicit_serialize
+class VampireMC(FiretaskBase):
+    """
+    If there is a suitable Heisenberg model, run Monte Carlo to compute the
+    critical temperature.
+
+    Args:
+        db_file (str): path to file containing the database credentials.
+        wf_uuid (int): Unique id for record keeping.
+        mc_settings (dict): A configuration dict for monte carlo.
+        average (bool): Only <J> exchange param.
+
+    TODO:
+        * Include HeisenbergModel convergence check.
+
+    """
+
+    required_params = [
+        "db_file",
+        "wf_uuid",
+        "mc_settings",
+        "average",
+    ]
+
+    def run_task(self, fw_spec):
+
+        db_file = env_chk(self["db_file"], fw_spec)
+        wf_uuid = self["wf_uuid"]
+        mc_settings = self["mc_settings"]
+        average = self["average"]
+
+        # Get Heisenberg models from db
+        mmdb = VaspCalcDb.from_db_file(db_file, admin=True)
+        mmdb.collection = mmdb.db["exchange"]
+
+        # Get documents
+        docs = list(
+            mmdb.collection.find(
+                {"wf_meta.wf_uuid": wf_uuid}, ["heisenberg_model", "nn_cutoff"]
+            )
+        )
+
+        hmodels = [HeisenbergModel.from_dict(d["heisenberg_model"]) for d in docs]
+        hmodel = hmodels[0]  # Take the model with smallest NN cutoff
+
+        # Vampire monte carlo settings
+        mc_box_size = mc_settings["mc_box_size"]
+        equil_timesteps = mc_settings["equil_timesteps"]
+        mc_timesteps = mc_settings["mc_timesteps"]
+
+        # Get a converged Heisenberg model if one was found
+        # if fw_spec["converged_heisenberg_model"]:
+        #     hmodel = HeisenbergModel.from_dict(fw_spec["converged_heisenberg_model"])
+
+        vc = VampireCaller(
+            mc_box_size=mc_box_size,
```

Comment:
Rather than expanding the mc_settings above, can you just do:

```python
vc = VampireCaller(
    hm=hmodel,
    avg=average,
    **mc_settings
)
```

That way if a user doesn't include one of the options, the VampireCaller will fall through back on the default settings.

Secondly, if you make `mc_settings` optional with a default value of an empty dict (`mc_settings = self.get("mc_settings, {})`)  this will again nicely support falling back on the default arguments.